rawhttp-core/src/test/kotlin/rawhttp/core/body/encoding/BodyDecodingTest.kt

```diff
@@ -125,4 +126,28 @@ class BodyDecodingTest {
         actualDecodedBody shouldBe "hello"
     }
 
+    @Test
+    fun corruptedChunkedGzippedBodyDoesNotHang() {
+        val gzippedFileStream = BodyDecodingTest::class.java.getResourceAsStream("corrupted-chunked-and-gzipped-response.http")
+        val response = RawHttp().parseResponse(gzippedFileStream)
+        try {
+            response.body.get().decodeBodyToString(StandardCharsets.UTF_8)
+        } catch(e: IOException) {}
+
+        true shouldBe true
```

Comment:
Thanks for the effort... It helps a lot that you've written a little test, but I have to say that the tests, as they are, are really not good enough... sorry to be blunt.

We need to check how long it took for the test to give up, what the Exception type was, what the message was...

I will accept your changes but will write more tests myself... I will push my code to this PR (hopefully, if I have permission) so we can discuss later if you would like.

Thanks again for finding the bug and contributing a first solution!

Reply:
Thank you for the input. You have permission to the repository.
About the thrown exception - this PR fixes two different scenarios, each throws a different exception.
1. In the chunked gzipped case, the piped input stream is stuck, so when we close it - it throws an IOException with the message "Pipe closed".
2. In the non chunked corrupted gzip, it throws an IOException with the message "Not in GZIP format" (with my patch, it throws it on finishDecoding(), previously it was suppressed)

I thought about wrapping the call to write() with a try/catch and then wrapping the exception with a different one - but I thought it would make the code more cumbersome (need to check for that specific message, and then assume that it is that edge case). Considering that, I thought that it is better as it is - but I guess opinions may vary on this issue. 
The test could be modified to expect the "pipe closed" - but I think that's an internal implementation issue. The main thing is that an IOException was thrown, and that the flow is not stuck.

Regarding how long it took for the test to give up - I'm adding code to test it. 

I'll also change the non-chunked corrupted gzip test to verify that the exception message is "Not in GZIP format"

If you have any more ideas for tests - go ahead. Of course.